### ISSUE TITLE
Feature: Markup changes to support accessibility

### ISSUE THREAD

**Issue by daviddurand** [Tizra/Tizra-Customer-Tracker#255]
We have too many h1 elements on navigational smart blocks, and we are missing some label tags that would tie captions to fields.

We will add a new "heading level" attribute to search blocks (and eventually all blocks that generate headings). This will set a minimum heading level for the content. Database defaults would ensure that existing designs will not change, but workbench defaults will guarantee that new "subordinate" blocks like search will be created with a default level level setting of 2, unless overriden by the designer. This level might apply to more than one heading level, so that a block that currently generates `h1` and `h2`  would, with a level setting of `3` generate `h3` and `h4` elements instead.

We will add `<label>` tags to input fields generally.
